### PR TITLE
Null safety

### PR DESCRIPTION
**Pull request by hunterbreathat-wk**
## Motivation
We need to migrate w_flux to null-safety, so that we may progress further on our migration story to Dart 3 / Null Safety.

## Changes
- Migrate to null safety
- Remove stable from testing matrix

#### Release Notes
- Migrate to null-safety

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Architecture member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/w_flux/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/w_flux/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: pubspec.yaml [hunk 1]
```diff
@@ -5,17 +5,27 @@ description: Flux library for uni-directional dataflow inspired by reflux and Fa
...
+
+dependency_overrides:
+  dart_dev:
+    git:
+      ref: fix_up_null_safety_hunter
+      url: https://github.com/Workiva/dart_dev
+  react:
+    git:
+      url: https://github.com/Workiva/react-dart
+      ref: null-safety-manual
```
Comment (hunterbreathat-wk) [inline pubspec.yaml:31]: These need to be removed before merge

File: example/pubspec.yaml [hunk 2]
```diff
@@ -20,5 +20,14 @@ dev_dependencies:
...
   w_flux:
     path: ../
+  react:
+    git:
+      url: https://github.com/Workiva/react-dart
+      ref: null-safety-manual
+  over_react:
+    git:
+      url: https://github.com/Workiva/over_react
+      ref: null-safety
```
Comment (hunterbreathat-wk) [inline example/pubspec.yaml:33]: Same here

File: .github/workflows/dart_ci.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        sdk: [ 2.18.7, stable ]
+        sdk: [ 2.18.7 ]
```
Comment (hunterbreathat-wk) [inline .github/workflows/dart_ci.yml:18]: Until we have full null-safety there is no reason to test on stable

File: lib/src/action.dart [hunk 4]
```diff
@@ -20,6 +20,16 @@ import 'package:w_common/disposable.dart';
...
 import 'package:w_flux/src/constants.dart' show v3Deprecation;
 
+class ActionOptionalPayload<T> extends Action<T?> {
+  @override
+  String get disposableTypeName => 'ActionOptionalPayload';
+
+  /// Dispatch this [Action] to all listeners. If a payload is supplied, it will
+  /// be passed to each listener's callback, otherwise null will be passed.
+  @override
+  Future call([T? payload]) => super.call(payload);
```
Comment (hunterbreathat-wk) [inline lib/src/action.dart:30]: Adding this `super` prevented stack overflow issue.

**Review by hunterbreathat-wk — COMMENTED** [review]
The remaining work and reasoning for some of the changes I made.

**Pull request closed without merging**